testsuite/MDAnalysisTests/core/test_atomgroup.py

```diff
@@ -190,6 +198,117 @@ def test_bogus_kwarg_pdb(self):
+        self.outfile = os.path.join(self.tempdir.name, 'writeatoms' + suffix)
+
+    def tearDown(self):
+        del self.universe
+        del self.tempdir
+
+    def universe_from_tmp(self):
+        return mda.Universe(self.outfile, convert_units=True)
+
+    def test_write_atoms(self):
+        self.universe.atoms.write(self.outfile)
+        u2 = self.universe_from_tmp()
+        assert_array_almost_equal(
+            self.universe.atoms.positions, u2.atoms.positions,
+            self.precision,
+            err_msg=("atom coordinate mismatch between original and {0!s} file"
+                     "".format(self.ext)))
+
+    def test_write_empty_atomgroup(self):
+        sel = self.universe.select_atoms('name doesntexist')
+        assert_raises(IndexError, sel.write, self.outfile)
+
+    def test_write_selection(self):
+        CA = self.universe.select_atoms('name CA')
+        CA.write(self.outfile)
+        u2 = self.universe_from_tmp()
+        # check EVERYTHING, otherwise we might get false positives!
+        CA2 = u2.atoms
+        assert_equal(len(u2.atoms), len(CA.atoms),
+                     "written CA selection does not match original selection")
+        assert_almost_equal(
+            CA2.positions, CA.positions, self.precision,
+            err_msg="CA coordinates do not agree with original")
+
+    def test_write_Residue(self):
+        G = self.universe.s4AKE.ARG[-2].atoms  # 2nd but last Arg
+        G.write(self.outfile)
+        u2 = self.universe_from_tmp()
+        # check EVERYTHING, otherwise we might get false positives!
+        G2 = u2.atoms
+        assert_equal(
+            len(u2.atoms), len(G.atoms),
+            "written R206 Residue does not match original ResidueGroup")
+        assert_almost_equal(
+            G2.positions, G.positions, self.precision,
+            err_msg="Residue R206 coordinates do not agree with original")
+
+    def test_write_ResidueGroup(self):
+        G = self.universe.s4AKE.LEU.atoms
+        G.write(self.outfile)
+        u2 = self.universe_from_tmp()
+        G2 = u2.atoms
+        assert_equal(
+            len(u2.atoms), len(G.atoms),
+            "written LEU ResidueGroup does not match original ResidueGroup")
+        assert_almost_equal(
+            G2.positions, G.positions, self.precision,
+            err_msg="ResidueGroup LEU coordinates do not agree with original")
+
+    def test_write_Segment(self):
+        G = self.universe.s4AKE.atoms
+        G.write(self.outfile)
+        u2 = self.universe_from_tmp()
+        G2 = u2.atoms
+        assert_equal(len(u2.atoms), len(G.atoms),
+                     "written s4AKE segment does not match original segment")
+        assert_almost_equal(
+            G2.positions, G.positions, self.precision,
+            err_msg="segment s4AKE coordinates do not agree with original")
+
+    def test_write_Universe(self):
+        U = self.universe
+        W = mda.Writer(self.outfile)
+        W.write(U)
+        W.close()
```

Comment:
Once again, probably an old test. Could use `with`, though, to avoid hanging files if the test fails.